### PR TITLE
feat: initial infinite scroll

### PR DESCRIPTION
**Pull request by cbt1**
Feature support in this PR:
- Scroll infinite* number of rows and/or columns (*if data contains more than 250k rows, show pagination in the footer)

Limitions:
- Only fetching data for cells that are visible to the user. Should consider using some kind of buffer to have cell data ready when scrolling in a resonable pace.
- None of the features in the "pagination table" support are supported yet. They will have to be added incrementally. Ex. selections, accessibility, right click..
- Is a stand-alone table, it should if possible be merged into the "pagination table" at some point. Possible when they are near feature parity.

### CODE REVIEW

File: src/table/Root.tsx [hunk 1]
```diff
@@ -24,6 +26,17 @@ export function render(props: RenderProps, reactRoot?: ReactDom.Root) {
   );
 }
 
+export function renderVirtualizedTable(props: WrapperProps, reactRoot?: ReactDom.Root) {
+  const muiTheme = muiSetup('ltr');
+
+  reactRoot?.render(
+    <React.StrictMode>
```
Comment (haxxmaxx) [inline src/table/Root.tsx:33]: FYI we scrapped strict mode since we got douple rerenders from useEffects I think it was

Comment (haxxmaxx) [reply src/table/Root.tsx:33]: https://medium.com/geekculture/the-tricky-behavior-of-useeffect-hook-in-react-18-282ef4fb570a Might be ways around it though

File: src/index.ts [hunk 2]
```diff
@@ -110,6 +129,7 @@ export default function supernova(env: Galaxy) {
               rect,
               footerContainer,
               announce,
+              model,
```
Comment (haxxmaxx) [inline src/index.ts:132]: why are you adding model here?

File: src/table/components/virtualized-table/constants/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const DEFAULT_ROW_HEIGHT = 32;
+
+const HEADER_HEIGHT = 32;
+
+const PAGINATION_HEIGHT = 49;
+
+const MAX_PAGE_SIZE = 250000;
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/constants/index.ts:7]: we probably want to make this dependent on font size later

**Review by haxxmaxx — COMMENTED** [review]
Pretty hard to review this much code. All of it looks good, but have not tried to understand the scrolling part hehe. We need to sync on how we should do styling. Maybe doesn't matter here, but it will down the line

File: src/table/components/virtualized-table/HeaderCell.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  style: React.CSSProperties;
+  data: {
+    columns: Column[];
+    headerStyle: GeneratedStyling;
+  };
+}
+
+const HeaderCell = ({ columnIndex, style, data }: HeaderCellProps) => {
+  const { columns, headerStyle } = data;
+  const datum = columns[columnIndex];
```
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/components/virtualized-table/HeaderCell.tsx:16]: Datum in this context means a single data point

File: src/index.ts (hunk 2 above)
Comment (cbt1) [reply src/index.ts:132]: That is likely some left over from the prototypes I made. Will remove it.

**Comment by cbt1** [comment]
> Pretty hard to review this much code. All of it looks good, but have not tried to understand the scrolling part hehe. We need to sync on how we should do styling. Maybe doesn't matter here, but it will down the line

Agree, it just ended up being pretty hard to have less code with an actual working scroll solution

File: src/table/Root.tsx (hunk 1 above)
Comment (cbt1) [reply src/table/Root.tsx:33]: I don't really have any issue with it being run in strict mode. It works as intended in dev env. So IMO I might as well keep it to potentially catch issues down the road.

File: src/index.ts [hunk 5]
```diff
@@ -82,14 +82,33 @@ export default function supernova(env: Galaxy) {
       const changeSortOrder = useSorting(model, layout.qHyperCube);
 
       const [pageInfo, setPageInfo] = useState(initialPageInfo);
+      const shouldRenderVirtualizedTable = layout.scrollType === 'virtualized';
       const [tableData] = usePromise(
         async () =>
-          env.carbon && !model?.getHyperCubeData
+          env.carbon || shouldRenderVirtualizedTable
```
Comment (haxxmaxx) [inline src/index.ts:88]: not sure what the consequenses of removing the sanity check for `getHyperCubeData` is here

File: src/table/Root.tsx (hunk 1 above)
Comment (haxxmaxx) [reply src/table/Root.tsx:33]: okok, seems to be in dev mode as well

File: src/index.ts [hunk 6]
```diff
@@ -82,14 +82,33 @@ export default function supernova(env: Galaxy) {
       const changeSortOrder = useSorting(model, layout.qHyperCube);
 
       const [pageInfo, setPageInfo] = useState(initialPageInfo);
+      const shouldRenderVirtualizedTable = layout.scrollType === 'virtualized';
```
Comment (haxxmaxx) [inline src/index.ts:85]: how is this set?

**Review by haxxmaxx — COMMENTED** [review]
Looks good in general, some more qustions

File: src/index.ts (hunk 5 above)
Comment (cbt1) [reply src/index.ts:88]: It should not have been remove. Will fix.

File: src/index.ts [hunk 7]
```diff
@@ -82,14 +82,33 @@ export default function supernova(env: Galaxy) {
       const changeSortOrder = useSorting(model, layout.qHyperCube);
 
       const [pageInfo, setPageInfo] = useState(initialPageInfo);
+      const shouldRenderVirtualizedTable = false; // layout.scrollMode === 1;
```
Comment (cbt1) [inline src/index.ts:85]: @haxxmaxx I ended up disabling this feature for now

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**